### DISCUSSION TITLE
CSS 선택자 네이밍 규칙 논의

### DISCUSSION THREAD

**Discussion by uraflower** [KakaoFunding/front-end#48] [Q&A]
## CSS 선택자 네이밍 규칙 논의
클래스명을 `기능_역할`로 작성한다는 컨벤션이 있긴 한데(ex. `modal_button`), 어떤 영역 자체를 가리키는 경우에는 약간 사용하기 애매할 수 있을 것 같아요.
예를 들어 `상품 목록 조회 페이지`에서 실제 상품/브랜드가 보여지는 영역을 가리켜야 할 때, `modal`처럼 기능으로 딱 떨어지는 게 아니니까 다른 이름이 또 필요할 것 같아요.

참고로 카카오선물하기에서 **영역에 대한** 선택자 네이밍은 이런 식인 듯 합니다.

- `article` 태그에는 `article` (클래스 말고 id로 쓰고 있어요)
- `section` 태그에는 `section` (ex. `section-theme`)
- `section` 태그가 아닌 어떤 영역은 `area` (ex. `area-category-list`)
- 스타일을 입히기 위해서 존재하는 클래스는 `wrap` (ex. `wrap-search-box`)
- 어떤 핵심 정보(ex. 상품 상세 조회 페이지 - 상세 내용)같은 경우에는 `cont` (contents 줄임말 같아요)
- 가끔 `view`를 쓰기도 함(ex. `view-buy-info`)

저희는 어떻게 할까요? 저는 이런 것들이 다 딱 와닿지가 않았는데 분류하고 나니 좀 나은 것 같긴 하네요.

**Comment by devkyoung2** [comment]
어떤곳에 section 태그를 쓰고 어떤곳에 article 태그를 써야하는지 잘 와닿지 않아요.... ㅠ.ㅠ
같은 의미에서 section 과 area, view 을 모두 사용하는건 조금 복잡할 것 같아요

그래서 저는 **영역**  을 가르키는 선택자는 area 어떤가요 ?

```
area-buy-info
area-product-detail
area-category-list
```

**Reply by uraflower** [reply]
생각해보니 `area`하나로 충분할 것 같네요..! 좋습니다!
